### PR TITLE
#176454681 add vendor certification for lawyer verification

### PR DESCRIPTION
**Pull request by tobieeniafe**
#### What does this PR do?
Add vendor certification for lawyer verification

#### Description of Task to be completed?
Add vendor certification for lawyer verification

#### How should this be manually tested?
`PUT /api/v1/user/vendor/certify`

#### What are the relevant pivotal tracker stories?
#176454681

### CODE REVIEW

File: server/services/user.service.js [hunk 1]
```diff
@@ -651,3 +651,35 @@ export const removeDirector = async ({ directorId, user }) => {
...
+  if (!user) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'User not found');
+  }
+
+  if (user.role !== USER_ROLE.VENDOR) {
+    throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, 'User is not a vendor');
+  }
+
+  if (user.vendor.verified !== true) {
+    throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, 'Vendor has not been verified');
```
Comment (andela-hpopoola) [inline server/services/user.service.js:667]: Might want to use the company name here too to make the message more personalized.


```suggestion
      throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, `${user.vendor.companyName} must be verified before approval as a certified vendor`);
```

File: server/services/user.service.js [hunk 2]
```diff
@@ -651,3 +651,35 @@ export const removeDirector = async ({ directorId, user }) => {
     throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error removing director', error);
   }
 };
+
+export const certifyVendor = async ({ vendorId, adminId }) => {
+  const user = await getUserById(vendorId);
+
+  if (!user) {
```
Comment (andela-hpopoola) [inline server/services/user.service.js:658]: Let's add a test for this 🙏🏽

File: server/services/user.service.js [hunk 3]
```diff
@@ -651,3 +651,35 @@ export const removeDirector = async ({ directorId, user }) => {
...
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'User not found');
+  }
+
+  if (user.role !== USER_ROLE.VENDOR) {
```
Comment (andela-hpopoola) [inline server/services/user.service.js:662]: Let's add a test for this too 🙏🏽

**Review by andela-hpopoola — CHANGES_REQUESTED** [review]
Thank you for working on this. I left a few comments to ensure we test for all edge cases.

File: server/services/user.service.js [hunk 4]
```diff
@@ -651,3 +651,35 @@ export const removeDirector = async ({ directorId, user }) => {
...
+    throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, 'User is not a vendor');
```
[the remark this answers is not in the document]
Comment (tobieeniafe) [reply server/services/user.service.js:663]: if the user role is not vendor, then the user can't have `user.vendor.companyName`

[the remark this answers is not in the document]
Comment (andela-hpopoola) [reply server/services/user.service.js:663]: That is so true 🤦🏽 .  Let's use `User is not a registered vendor`

File: test/controllers/user.controller.test.js [hunk 5]
```diff
@@ -2252,22 +2257,24 @@ describe('User Controller', () => {
...
 
       context('when vendor has not been verified', () => {
-        const unverifiedVendorId = mongoose.Types.ObjectId();
-        const unverifiedVendor = UserFactory.build({
-          _id: unverifiedVendorId,
-          role: USER_ROLE.VENDOR,
-          activated: true,
-          vendor: {
-            verified: false,
+        const unverifiedVendor = UserFactory.build(
```
Comment (andela-hpopoola) [inline test/controllers/user.controller.test.js:2267]:
```suggestion
        const unVerifiedVendor = UserFactory.build(
```

**Review by andela-hpopoola — APPROVED** [review]
Looks good to me. Thank you for your hard work on this. I left 2 non-blocking comments, feel free to merge after resolving.
